google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -375,37 +377,34 @@ private PsiElement generateFromTemplate(String templateName, String templateDire
           .findDirectory(virtualFile);
 
       if (directory != null) {
-        ApplicationManager.getApplication().runWriteAction((Computable<PsiElement>) () -> {
-          PsiDirectory configDirectory;
-          try {
-            directory.checkCreateSubdirectory(templateDirectoryName);
-            configDirectory = directory.createSubdirectory(templateDirectoryName);
-          } catch (IncorrectOperationException ioe) {
-            // checkCreateSubdirectory threw an exception suggesting that the directory may already
-            // exist. Skip creating the directory and attempt to write file.
-            configDirectory = directory.findSubdirectory(templateDirectoryName);
-          }
+        PsiDirectory configDirectory;
+        try {
+          directory.checkCreateSubdirectory(templateDirectoryName);
+          configDirectory = directory.createSubdirectory(templateDirectoryName);
+        } catch (IncorrectOperationException ioe) {
+          // checkCreateSubdirectory threw an exception suggesting that the directory may already
+          // exist. Skip creating the directory and attempt to write file.
+          configDirectory = directory.findSubdirectory(templateDirectoryName);
+        }
 
-          if (configDirectory != null
-              && FileTemplateUtil.canCreateFromTemplate(
-              new PsiDirectory[]{configDirectory}, configTemplate)) {
-            try {
-              return FileTemplateUtil.createFromTemplate(
-                  configTemplate,
-                  fileName,
-                  templateProperties,
-                  configDirectory);
-            } catch (Exception e) {
-              // If the file already exists, this exception will be thrown by createFromTemplate
-              // We want to silently skip the generation in this case.
-              logger.debug("Failed to create app yaml from template. " + e.getMessage());
-            }
+        if (configDirectory != null
```

Comment:
to elaborate, that condition is checking 2 things:

1) that the directory exists: the code above ensures that the directory is created ore retrieved if it already existed -> this would only fail for some unexpected IO error
2) that the template can be created -> this is static and not based on any user action so should really never happen

It probably makes sense to raise an ide level error in this case